src/Type/Php/SprintfFunctionDynamicReturnTypeExtension.php

```diff
@@ -125,32 +125,45 @@ public function getTypeFromFunctionCall(
 			return $singlePlaceholderEarlyReturn;
 		}
 
+		if ($allPatternsNonFalsy) {
+			return new IntersectionType([
+				new StringType(),
+				new AccessoryNonFalsyStringType(),
+			]);
+		}
+
 		$isNonEmpty = $allPatternsNonEmpty;
 		if (
-			count($formatStrings) === 0
+			!$isNonEmpty
 			&& $functionReflection->getName() === 'sprintf'
-			&& count($args) === 2
+			&& count($args) >= 2
 			&& $formatType->isNonEmptyString()->yes()
-			&& $scope->getType($args[1]->value)->isNonEmptyString()->yes()
 		) {
-			$isNonEmpty = true;
+			$allArgsNonEmpty = true;
+			foreach ($args as $key => $arg) {
+				if ($key === 0) {
+					continue;
+				}
+
+				if (!$scope->getType($arg->value)->toString()->isNonEmptyString()->yes()) {
```

Comment:
The toString was added for things like IntegerRange which need to be casted before the isNonEmptyString check.